tests/api/Jalasoft.TeamUp.Projects.API.Tests/PatchInvitationByIdTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace Jalasoft.TeamUp.Projects.API.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using Jalasoft.TeamUp.Projects.API.Controllers;
+    using Jalasoft.TeamUp.Projects.API.Tests.Utils;
+    using Jalasoft.TeamUp.Projects.Core.Interfaces;
+    using Jalasoft.TeamUp.Projects.Models;
+    using Jalasoft.TeamUp.Projects.ProjectsException;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Moq;
+    using Xunit;
+
+    public class PatchInvitationByIdTests
+    {
+        private readonly Mock<IInvitationsService> mockService;
+        private readonly DefaultHttpContext mockHttpContext;
+        private readonly PatchInvitationById updateInvitation;
+
+        public PatchInvitationByIdTests()
+        {
+            this.mockService = new Mock<IInvitationsService>();
+            this.mockHttpContext = new DefaultHttpContext();
+            this.updateInvitation = new PatchInvitationById(this.mockService.Object);
+        }
+
+        [Theory]
+        [InlineData(@"[
+        {""op"" : ""replace"", ""path"" : ""/status"", ""value"" : ""Accepted""}
+        ]")]
+        public async void PatchInvitationById_IdIsValid_OkObjectResult(string body)
+        {
+            Invitation stubInvitation = StubInvitation.GetStubInvitation();
+
+            HttpRequest request = this.mockHttpContext.Request;
+            request.Body = SetStream.Setstream(body);
+            this.mockService.Setup(service => service.GetInvitation(new Guid("5a7939fd-59de-33bd-a092-f5d8434584de"))).Returns(stubInvitation);
+            this.mockService.Setup(service => service.UpdateInvitation(null)).Returns(stubInvitation);
+            var response = await this.updateInvitation.UpdateInvitation(request, new Guid("5a7939fd-59de-33bd-a092-f5d8434584de"));
+            var okObjectResult = Assert.IsType<OkObjectResult>(response);
+
+            Assert.Equal(200, okObjectResult.StatusCode);
+        }
+
+        [Theory]
+        [InlineData(@"[
+        {""op"" : ""replace"", ""path"" : ""/status"", ""value"" : ""Accepted""}
+        ]")]
+        public async void PatchInvitationById_IdIsNotValid_NotFoundResult(string body)
+        {
+            HttpRequest request = this.mockHttpContext.Request;
+            request.Body = SetStream.Setstream(body);
+            this.mockService.Setup(service => service.GetInvitation(Guid.Parse("5a7939fd-59de-44bd-a092-f5d8434584de"))).Throws(new ProjectsException(ProjectsErrors.NotFound));
```

Comment:
the service does not return an exception when the id is not found it only returns null, the exception is handled in the API